chassis/core/src/main/java/com/griddynamics/jagger/rest/MasterRestController.java

```diff
@@ -0,0 +1,188 @@
+            taskInfos.add(getTaskInfoFrom(task));
+        }
+        
+        return new ResponseEntity<>(taskInfos, HttpStatus.OK);
+    }
+    
+    @GetMapping(path = "/tasks/{name}")
+    public ResponseEntity<TaskInfo> getTaskInfo(@PathVariable String name) {
+        
+        if (nameToTaskMap == null) {
+            nameToTaskMap = mapTaskToName(configuration.getTasks());
+        }
+        
+        Task task = nameToTaskMap.get(name);
+        if (task == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(getTaskInfoFrom(task), HttpStatus.OK);
+    }
+    
+    private Map<String, Task> mapTaskToName(List<? extends Task> taskList) {
+        Map<String, Task> nameToTaskMap = Maps.newHashMap();
+        for (Task task : taskList) {
+            nameToTaskMap.put(task.getTaskName(), task);
+            if (task instanceof CompositeTask) {
+                CompositeTask compositeTask = (CompositeTask) task;
+                nameToTaskMap.putAll(mapTaskToName(compositeTask.getLeading()));
+                nameToTaskMap.putAll(mapTaskToName(compositeTask.getAttendant()));
+            }
+        }
+        
+        return nameToTaskMap;
+    }
+    
+    private TaskInfo getTaskInfoFrom(Task task) {
+        TaskInfo taskInfo = new TaskInfo();
+        if (task instanceof CompositeTask) {
+            CompositeTask compositeTask = (CompositeTask) task;
+            CompositeTaskInfo compositeTaskInfo = new CompositeTaskInfo();
+            
+            compositeTaskInfo.leadingTasks = Lists.newArrayList();
+            for (Task leadingTask : compositeTask.getLeading()) {
+                compositeTaskInfo.leadingTasks.add(getTaskInfoFrom(leadingTask));
+            }
+            
+            compositeTaskInfo.attendantTasks = Lists.newArrayList();
+            for (Task attendantTask : compositeTask.getAttendant()) {
+                compositeTaskInfo.attendantTasks.add(getTaskInfoFrom(attendantTask));
+            }
+            
+            taskInfo = compositeTaskInfo;
+        }
+        
+        taskInfo.taskName = task.getTaskName();
+        taskInfo.sessionId = sessionIdProvider.getSessionId();
+        if (task.getNumber() > 0) {  // 0 means it wasn't set yet.
+            taskInfo.number = task.getNumber();
+            taskInfo.taskId = taskIdProvider.stringify(task.getNumber());
+            taskInfo.status = taskExecutionStatusProvider.getStatus(taskInfo.taskId);
+        }
+        
+        return taskInfo;
+    }
+    
+    public static class CompositeTaskInfo extends TaskInfo {
+        private List<TaskInfo> leadingTasks;
+        private List<TaskInfo> attendantTasks;
+        
+        public List<TaskInfo> getLeadingTasks() {
+            return leadingTasks;
+        }
+        
+        public List<TaskInfo> getAttendantTasks() {
+            return attendantTasks;
+        }
+    }
+    
+    public static class TaskInfo {
```

Comment:
we need to review model for task and session (what information should they contain). we already have several implementations of models for these entities